<!--REQUEST_ID=0fb064f0-0cfa-4528-96f8-3a0ab5f37bab-->
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46619&attempt_id=7efc34bf-4529-4137-aee2-7022adf18900)

Attempted: papi

<details><summary>Partial log (click to expand)</summary><p>

```
clang -I../testlib -I.. -I. -DNEED_FFSLL -O2  -c display_error.c
clang -I../testlib -I.. -I. -DNEED_FFSLL -O2  -c instructions_testcode.c
clang -o cycles_validation cycles_validation.o ../testlib/libtestlib.a display_error.o instructions_testcode.o ../libpapi.a  -lrt
ld: library not found for -lrt
clang-5.0: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [Makefile.recipies:43: cycles_validation] Error 1
make[1]: Leaving directory '/private/tmp/nix-build-papi-5.6.0.drv-0/icl-papi-31d2ae8200ba/src/validation_tests'
make: *** [Makefile.inc:237: validation_tests] Error 2
builder for '/nix/store/sf5pksvfa18cgpizbjilm6krkx1p00yz-papi-5.6.0.drv' failed with exit code 2
error: build of '/nix/store/sf5pksvfa18cgpizbjilm6krkx1p00yz-papi-5.6.0.drv' failed
```
</p></details>

--- a/pkgs/development/libraries/science/benchmark/papi/default.nix
+++ b/pkgs/development/libraries/science/benchmark/papi/default.nix
@@ -1,0 +1,30 @@
+{ stdenv
+, fetchurl
+}:
+
+stdenv.mkDerivation rec {
+  version = "5.6.0";
+  name = "papi-${version}";
+
+  src = fetchurl {
+    url = "https://bitbucket.org/icl/papi/get/papi-5-6-0-t.tar.gz";
+    sha256 = "13mngf9kl0y2wfxqvkad0smdaag7k8fvw82b4312gx62nwhc1i6r";
+  };
+
+  buildInputs = [ stdenv ];
+
+  preConfigure = ''
+    cd src
+  '';
+
+  doCheck = true;
+  checkTarget = "test";
+
+  meta = with stdenv.lib; {
+    homepage = https://icl.utk.edu/papi/;
+    description = "PAPI provides the tool designer and application engineer with a consistent interface and methodology for use of the performance counter hardware found in most major microprocessors";
+    license = licenses.bsdOriginal;
+    platforms = platforms.linux;
+    maintainers = [ maintainers.costrouc ];
+  };
+}

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -20590,6 +20590,10 @@ with pkgs;
   tetgen = callPackage ../applications/science/geometry/tetgen { }; # AGPL3+
   tetgen_1_4 = callPackage ../applications/science/geometry/tetgen/1.4.nix { }; # MIT
 
+  ### SCIENCE/BENCHMARK
+
+  papi = callPackage ../development/libraries/science/benchmark/papi { };
+
   ### SCIENCE/BIOLOGY
 
   alliance = callPackage ../applications/science/electronics/alliance { };

